engine/src/cython/io.cpp

```diff
@@ -47,41 +66,65 @@ TableSchema parseSchema(std::vector<std::string> files,
 		parser = std::make_shared<ral::io::json_parser>(args_map);
 	} else if(fileType == ral::io::DataType::CSV) {
 		parser = std::make_shared<ral::io::csv_parser>(args_map);
-	}
+	} else if(fileType == ral::io::DataType::MYSQL) {
+#ifdef MYSQL_SUPPORT
+		parser = std::make_shared<ral::io::mysql_parser>();
+    auto sql = ral::io::getSqlInfo(args_map);
+    provider = std::make_shared<ral::io::mysql_data_provider>(sql);
+#endif
+    isSqlProvider = true;
+  } else if(fileType == ral::io::DataType::SQLITE) {
+#ifdef SQLITE_SUPPORT
+//		parser = std::make_shared<ral::io::sqlite_parser>();
+//    auto sql = ral::io::getSqlInfo(args_map);
+//    provider = std::make_shared<ral::io::sqlite_data_provider>(sql.schema,
+//                                                               sql.table,
+//                                                               sql.table_filter,                                                          sql.table_batch_size);
+//    isSqlProvider = true;
+#endif
+  }
+
+  if (!isSqlProvider) {
+      provider = std::make_shared<ral::io::uri_data_provider>(uris, ignore_missing_paths);
+  }
 
-	std::vector<Uri> uris;
-	for(auto file_path : files) {
-		uris.push_back(Uri{file_path});
-	}
-	auto provider = std::make_shared<ral::io::uri_data_provider>(uris, ignore_missing_paths);
 	auto loader = std::make_shared<ral::io::data_loader>(parser, provider);
 
 	ral::io::Schema schema;
 
 	try {
-
 		bool got_schema = false;
-		while (!got_schema && provider->has_next()){
-			ral::io::data_handle handle = provider->get_next();
-			if (handle.file_handle != nullptr){
-				parser->parse_schema(handle.file_handle, schema);
-				if (schema.get_num_columns() > 0){
-					got_schema = true;
-					schema.add_file(handle.uri.toString(true));
-				}
-			}
+    if (isSqlProvider) {
+        ral::io::data_handle handle = provider->get_next(false);
+        parser->parse_schema(handle, schema);
+        if (schema.get_num_columns() > 0){
+          got_schema = true;
+        }
+    } else {
+      while (!got_schema && provider->has_next()){
+        ral::io::data_handle handle = provider->get_next();
+        if (handle.file_handle != nullptr){
+          parser->parse_schema(handle, schema);
+          if (schema.get_num_columns() > 0){
+            got_schema = true;
+            schema.add_file(handle.uri.toString(true));
+          }
+        }
+      }
 		}
 		if (!got_schema){
 			std::cout<<"ERROR: Could not get schema"<<std::endl;
 		}
 
 		bool open_file = false;
-		while (provider->has_next()){
-			std::vector<ral::io::data_handle> handles = provider->get_some(64, open_file);
-			for(auto handle : handles) {
-				schema.add_file(handle.uri.toString(true));
-			}
-		}
+    if (!isSqlProvider) {
+      while (provider->has_next()){
+        std::vector<ral::io::data_handle> handles = provider->get_some(64, open_file);
```

Comment:
@williamBlazing it seems to be used here the abstractsql_dat_provider provides this method so its not just decorative